pallets/pool-system/src/tests/ratios.rs

```diff
@@ -0,0 +1,404 @@
+			.unwrap()
+			.tranches
+			.residual_top_slice()
+			.iter()
+			.for_each(|tranche| {
+				assert_eq!(tranche.ratio, Perquintill::from_percent(100));
+			});
+
+		// Attempt to redeem 40%
+		assert_ok!(Investments::update_redeem_order(
+			RuntimeOrigin::signed(0),
+			TrancheCurrency::generate(0, JuniorTrancheId::get()),
+			200 * CURRENCY
+		));
+		assert_ok!(PoolSystem::close_epoch(pool_owner_origin.clone(), 0));
+		assert_ok!(Investments::collect_redemptions(
+			RuntimeOrigin::signed(0),
+			TrancheCurrency::generate(0, JuniorTrancheId::get()),
+		));
+
+		// Ensure ratio is 100
+		Pool::<Runtime>::get(0)
+			.unwrap()
+			.tranches
+			.residual_top_slice()
+			.iter()
+			.for_each(|tranche| {
+				assert_eq!(tranche.ratio, Perquintill::from_percent(100));
+			});
+
+		// Attempt to redeem everything
+		assert_ok!(Investments::update_redeem_order(
+			RuntimeOrigin::signed(0),
+			TrancheCurrency::generate(0, JuniorTrancheId::get()),
+			300 * CURRENCY
+		));
+		assert_ok!(PoolSystem::close_epoch(pool_owner_origin.clone(), 0));
+		assert_ok!(Investments::collect_redemptions(
+			RuntimeOrigin::signed(0),
+			TrancheCurrency::generate(0, JuniorTrancheId::get()),
+		));
+
+		// Ensure ratio is 0
+		Pool::<Runtime>::get(0)
+			.unwrap()
+			.tranches
+			.residual_top_slice()
+			.iter()
+			.for_each(|tranche| {
+				assert_eq!(tranche.ratio, Perquintill::from_percent(0));
+			});
+
+		// Ensure ratio goes up again
+		invest_close_and_collect(0, vec![(0, JuniorTrancheId::get(), 300 * CURRENCY)]);
+
+		// Ensure ratio 100
+		Pool::<Runtime>::get(0)
+			.unwrap()
+			.tranches
+			.residual_top_slice()
+			.iter()
+			.for_each(|tranche| {
+				assert_eq!(tranche.ratio, Perquintill::from_percent(100));
+			});
+	});
+}
+
+#[test]
+fn ensure_ratios_are_distributed_correctly_3_tranches() {
+	new_test_ext().execute_with(|| {
+		let pool_owner = DEFAULT_POOL_OWNER;
+		let pool_owner_origin = RuntimeOrigin::signed(pool_owner);
+
+		util::default_pool::create_with_tranche_input(util::default_pool::three_tranche_input());
+
+		// Assert ratios are all zero
+		Pool::<Runtime>::get(0)
+			.unwrap()
+			.tranches
+			.residual_top_slice()
+			.iter()
+			.for_each(|tranche| {
+				assert_eq!(tranche.ratio, Perquintill::from_percent(0));
+			});
+
+		// Force min_epoch_time to 0 without using update
+		// as this breaks the runtime-defined pool
+		// parameter bounds and update will not allow this.
+		Pool::<Runtime>::try_mutate(0, |maybe_pool| -> Result<(), ()> {
+			maybe_pool.as_mut().unwrap().parameters.min_epoch_time = 0;
+			maybe_pool.as_mut().unwrap().parameters.max_nav_age = u64::MAX;
+			Ok(())
+		})
+		.unwrap();
+
+		invest_close_and_collect(
+			0,
+			vec![
+				(0, JuniorTrancheId::get(), 500 * CURRENCY),
+				(0, SeniorTrancheId::get(), 500 * CURRENCY),
+				(0, SecondSeniorTrancheId::get(), 500 * CURRENCY),
+			],
+		);
+
+		let check_ratios = [
+			Perquintill::from_rational_with_rounding(1u64, 3, Rounding::Up).unwrap(),
```

Comment:
NIT: Any reason why this diverges from the lines below?
```suggestion
			Perquintill::from_rational(1u64, 3),
```

Reply:
`Perquintill` usually rounds down, but we need to have the round up version here as the residual takes the rest.